Automattic/wp-calypso
# Checkout: Remove useShowAddCouponSuccessMessage

#### Changes proposed in this Pull Request

There have been several bugs in how the shopping-cart endpoint sets the `is_coupon_applied` property and when it adds coupon success or failure messages to the `ResponseCart` object. Those should all be fixed by D53596-code. However, because of the unreliability of those messages in the past, new checkout includes a special hook, `useShowAddCouponSuccessMessage`, which attempts to show a coupon success message when one _should_ be shown by the server. That hook doesn't have as much information as the server, and so it can sometimes be wrong and display something like `The '' coupon was successfully applied to your shopping cart.`.

With the server fix applied, this hook is no longer necessary and we can just remove it. The server will take care of messages.

#### Testing instructions

- Apply D53596-code.
- Add a product to your cart and visit checkout.
- Click to add a coupon to your cart and enter an invalid coupon like "asdioadjaiodjwaojdawd". Verify that you see an error message that says the coupon failed.
- Click to add a coupon to your cart and enter a valid coupon. Verify that you see an success message that says the coupon succeeded.
- While the valid coupon is applied, click to change the cart in some way; for example, change the plan term length or remove a product. Verify that you see no messages about coupons and that the applied coupon remains.
- While the valid coupon is applied, make the coupon invalid somehow; for example, delete the coupon on the server (only delete coupons you created!). Then click to change the cart in some way as above. Verify that you see no messages about coupons and that the applied coupon vanishes.
- Add a valid coupon to the cart again and then click to remove the coupon from the cart. Verify that you see an success message that says the coupon was removed.
